src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            size = addEntry(Objects.hashCode(key), key, value, getIndex(key)) ? ++size : size;
+        }
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Object key) {
+        return (key == null) ? 0 : Objects.hashCode(key) & table.length - 1;
+    }
+
+    private void resize() {
+        if (size >= threshold) {
+            size = 0;
+            threshold = table.length << 1;
+            Node<K, V>[] oldTable = table;
+            table = new Node[table.length << 1];
+            transfer(oldTable);
+        }
+    }
+
+    private void transfer(Node<K, V>[] table) {
+        for (Node<K, V> kvNode : table) {
+            while (kvNode != null) {
+                put(kvNode.key, kvNode.value);
+                kvNode = kvNode.next;
+            }
+        }
+    }
+
+    private void putForNullKey(V value) {
```

Comment:
Let's get rid of this method. In the put() method you can check if the "bucket" is empty, and if it's null, you just add an node to it.